code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/builders/InputBoxNotificationBuilder.kt

```diff
@@ -0,0 +1,302 @@
+
+    @Throws(NotificationConstructionFailedException::class)
+    fun construct(
+        context: Context,
+        pushTemplate: InputBoxPushTemplate,
+        trackerActivityClass: Class<out Activity>?,
+        broadcastReceiverClass: Class<out BroadcastReceiver>?
+    ): NotificationCompat.Builder {
+        val cacheService = ServiceProvider.getInstance().cacheService
+            ?: throw NotificationConstructionFailedException("Cache service is null, input box push notification will not be constructed.")
+
+        Log.trace(
+            PushTemplateConstants.LOG_TAG,
+            SELF_TAG,
+            "Building an input box template push notification."
+        )
+        val packageName = context.packageName
+        val smallLayout = RemoteViews(packageName, R.layout.push_template_collapsed)
+        val expandedLayout = RemoteViews(packageName, R.layout.push_template_expanded)
+
+        // create a silent notification channel if needed
+        if (pushTemplate.isFromIntent == true && Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val notificationManager =
+                context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+            AEPPushNotificationBuilder.setupSilentNotificationChannel(
+                notificationManager,
+                pushTemplate.getNotificationImportance()
+            )
+        }
+
+        // create the notification channel if needed
+        val channelIdToUse = AEPPushNotificationBuilder.createChannelIfRequired(
+            context,
+            pushTemplate.channelId,
+            pushTemplate.sound,
+            pushTemplate.getNotificationImportance()
+        )
+
+        // create the notification builder with the common settings applied
+        val notificationBuilder = AEPPushNotificationBuilder.construct(
+            context,
+            pushTemplate,
+            channelIdToUse,
+            trackerActivityClass,
+            smallLayout,
+            expandedLayout,
+            R.id.basic_expanded_layout
+        )
+
+        // get push payload data. if we are handling an intent then we know that we should be building a feedback received notification.
+        val imageUri =
+            if (pushTemplate.isFromIntent != true) pushTemplate.imageUrl else pushTemplate.feedbackImage
+        val pushImage = PushTemplateImageUtil.downloadImage(cacheService, imageUri)
+
+        if (pushImage != null) {
+            expandedLayout.setImageViewBitmap(R.id.expanded_template_image, pushImage)
+        } else {
+            Log.trace(
+                PushTemplateConstants.LOG_TAG,
+                SELF_TAG,
+                "No image found for input box push template."
+            )
+            expandedLayout.setViewVisibility(R.id.expanded_template_image, View.GONE)
+        }
+
+        val bodyText =
+            if (pushTemplate.isFromIntent != true) pushTemplate.body else pushTemplate.feedbackText
```

Comment:
nit: `if (pushTemplate.isFromIntent == true) pushTemplate.feedbackText else pushTemplate.body` for readability